T0021 [IMP] A few corrections for the sub_management module

ecino opened the pull request.

ecino committed:
T0021 [IMP] A few corrections for the sub_management module

NoeBerdoz commented on sponsorship_sub_management/__manifest__.py:
```diff
@@ -36,9 +36,7 @@
     "depends": [
-        "sponsorship_compassion",
-        "base_automation",
-        "crm_compassion"
```
Removing crm_compassion from dependencies will cause issue when installing and using the module on an instance that doesn't have it installed as it is used in the contract model on this function [def onchange_child_id(self)](https://github.com/CompassionCH/compassion-modules/blob/165713ac38fd81ba999fd9ab138bf2a7d42f074b/sponsorship_sub_management/models/contracts.py#LL116C18-L116C18) on super().onchange_child_id()

ecino replied:
Good catch. However the dependency was not necessary so I adapted the onchange method so that it's not the same as the one in crm_compassion module. If we can remove dependencies it's always good.

NoeBerdoz commented on sponsorship_sub_management/__manifest__.py:
```diff
@@ -36,9 +36,7 @@
     "website": "https://www.compassion.ch",
     "depends": [
-        "sponsorship_compassion",
-        "base_automation",
```
The module is used with 3 automated actions
- Sponsorship SUB Proposal
- Sponsorship SUB Proposal validation
- Check SUB Accept/Reject after 50 days

I think we need these for the module to work correctly, and the automated actions are depending on base_automation right?

ecino replied:
Module sponsorship_compassion is already depending on base_automation through recurring_contract, so we don't need to specify again the dependency.

NoeBerdoz reviewed (commented):
2 comments about the dependencies, otherwise all good thank you

ecino committed:
[FIX] Performance issues in crm_compassion

- Remove origin in holds which was used only at one place. Replace it with a function
- Add some indexes on hold objects
- Install still takes long because of the event_id field added on move lines
